jupyterhub/auth.py

```diff
@@ -746,19 +749,19 @@ def authenticate(self, handler, data):
         return username
 
 
-    @run_on_executor
-    def pre_spawn_start(self, user, spawner):
+    def pre_exec_fn(self, user, spawner):
         """Open PAM session for user if so configured"""
         if not self.open_sessions:
             return
         try:
             pamela.open_session(user.name, service=self.service, encoding=self.encoding)
         except pamela.PAMError as e:
             self.log.warning("Failed to open PAM session for %s: %s", user.name, e)
```

Comment:
Since this is now called in the subprocess, I'm not sure `self.log.warning` is going to work. Probably switch this to a lower-level print to stderr.

Reply:
I've confirmed it does work in a subprocess.